.github/workflows/buildcheck.yml

```diff
@@ -20,21 +21,31 @@ jobs:
     needs: call-reusable-workflow
     runs-on: ubuntu-latest
     steps:
-    - name: Restore cached tags and jars
-      id: cache-tags-jars
-      uses: actions/cache@v4
+    - name: Set up JDK for x64
+      uses: actions/setup-java@v4
       with:
-        path: |
-          ./cached-tags
-          jameica
-          hibiscus
-        key: ${{ runner.os }}-${{ github.repository_id }}
-
+        java-version: '17'
+        distribution: 'temurin'
+        architecture: x64
     - name: Checkout openjverein
       uses: actions/checkout@v4
       with:
         path: jverein
 
+    - name: Restore build dependencies
+      id: restore-cache
+      uses: actions/cache/restore@v4
+      with:
+        path: |
+          jameica/bin/
+          jameica/lib/
+          jameica/releases/jameica-lib.jar
+          hibiscus/bin/
+          hibiscus/lib/
+          hibiscus/releases/hibiscus-lib.jar
```

Comment:
Das sind die Abhängigkeiten aus dem ant build

Reply:
Die natürlich nun wieder redundant in allen Workflows drin stehen.

Reply:
Das geht meines Wissens nicht anders. path ist ein mandatory argument.

Reply:
Du kannst den kompletten Step nach `reusable-build.yml` auslagern und dann einfach hier verwenden. So hat es ja vorher auch mit dem Step "cache-tags-jars" funktioniert.

Reply:
Wie soll das funktionieren? Vorher gab es in reusable-build.yml zwei cache steps um zu Laden und bei Änderung zu den cache neu zu erstellen und in jedem "echten" build job wurde der cache step als allererstes aufgerufen.  Die Jobs sind unabhängig voneinander und übertragen keinen Zustand.

Reply:
Mit `needs: call-reusable-workflow` wird der Workflow `./.github/workflows/reusable-build.yml` ausgeführt. Wird exakt das ausgeführt, was hier bis Zeile 48 definiert ist. Daher ist es hier nicht notwendig, es nochmal auszuführen und auch nicht in den anderen Workflows. Das ist ja die Grundidee der reusable workflows.

Reply:
`needs:` führt dazu, dass die Jobs nacheinander laufen. Sie laufen aber nicht im gleichen Kontext. Ich kann's gerne mal abbrechen lassen.

Reply:
Du kannst es dir gerne anschauen https://github.com/tobidope/jverein/actions/runs/13422033743

```
compile:
    [mkdir] Created dir: /home/runner/work/jverein/jverein/jverein/bin
    [javac] Compiling 868 source files to /home/runner/work/jverein/jverein/jverein/bin

BUILD FAILED
/home/runner/work/jverein/jverein/jverein/build/build.xml:57: /home/runner/work/jverein/jverein/hibiscus/lib does not exist.

Total time: 0 seconds
```

Reply:
Wenn es da nicht noch einen Schalter gibt, bleibe ich dabei. Jobs laufen in unabhängigen Umgebungen. Ein Job kann auf Ausgaben eines anderen Jobs nur durch explizite Datenübergabe (Cache, Artifacts, Variablen) zugreifen, es bleibt nichts im Dateisystem stehen.

Reply:
Stimmt, du hast recht. An die Post- und Complete-Steps habe nicht mehr gedacht. Die räumen wieder auf. Da ich das mit dem "Cache Reuse" ohne Code Duplication jedoch schon mal irgendwo gesehen habe, habe ich nochmal kurz recherchiert. Das wurde mit [Composite Actions](https://docs.github.com/en/actions/sharing-automations/creating-actions/creating-a-composite-action) gemacht. Kennst du die?
Ich weiß, es geht hier nur um ca. 25 Zeilen Code, allerdings in vier Kopien.

Reply:
Hier werden die drei grundsätzlichen Möglichkeiten aufgezeigt: https://stackoverflow.com/a/71570847